internal/config/report.go

```diff
@@ -50,30 +61,62 @@ func NewReportConfig(c *cli.Context) ReportConfig {
 
 // Init initializes the ReportConfig
 func (c *ReportConfig) Init(logger *zap.SugaredLogger) (err error) {
-	if c.Template != "" {
-		if c.Format == "" {
+	formats := make(map[string]MappedFormat)
+	var format, template []string
+	var output []*os.File
+
+	if c.format == "" {
+		if c.template != "" {
 			logger.Warn("--template is ignored because --format template is not specified. Use --template option with --format template option.")
-		} else if c.Format != "template" {
-			logger.Warnf("--template is ignored because --format %s is specified. Use --template option with --format template option.", c.Format)
 		}
+		formats["table"] = DefaultFormat
+	} else {
+		format = strings.Split(c.format, ",")
+		output, err = c.mapOutput(format, logger)
+		if err != nil {
+			return err
+		}
+		template = c.mapTemplate(format, logger)
 	}
-	if c.Format == "template" && c.Template == "" {
-		logger.Warn("--format template is ignored because --template not is specified. Specify --template option when you use --format template.")
+
+	for i, f := range format {
+		_, ok := formats[f]
+		if ok {
+			logger.Warnf("--format %s is ignored because it has been specified.", f)
+			continue
+		}
+
+		if format[i] == "template" {
+			if template == nil {
+				logger.Warn("--format template is ignored because --template is not specified. Specify --template option when you use --format template.")
+				continue
+			} else {
+				formats[f] = MappedFormat{
+					Output:   output[i],
+					Template: template[0],
+				}
+			}
+		} else {
+			formats[f] = MappedFormat{
+				Output:   output[i],
+				Template: "",
+			}
+		}
```

Comment:
I believe we can make it simpler. Please feel free to ask me if there are difficult specifications to simplify the source code. We might be able to change it.

Reply:
refactored this into another function